setup.py

```diff
@@ -14,12 +14,12 @@
     platforms='linux',
     classifiers=[
         'License :: OSI Approved :: MIT License',
-        'Programming Language :: Python :: 2',
-        'Programming Language :: Python :: 2.7',
         'Programming Language :: Python :: 3',
         'Programming Language :: Python :: 3.4',
         'Programming Language :: Python :: 3.5',
+        'Programming Language :: Python :: 3.6',
     ],
+    python_requires='>=3.3',
```

Comment:
@chriskuehl you suggested `>=3.7` in https://github.com/Yelp/aactivator/issues/42#issuecomment-1432304271, but `shlex.quote` was added in 3.3, so I think `>=3.3` should be sufficient?

Reply:
3.3 is ancient, please us 3.7

Reply:
Bionic has python3.6 (https://packages.ubuntu.com/bionic/python3), and hasn't reached the end of standard support (yet). Nothing in the script requires anything > 3.3. Would there be a problem with requiring >=3.3 instead of >=3.7?

Reply:
both of us have asked you to use 3.7, why are you fighting it. everything older than that is end of lifed

Reply:
The reason I suggested 3.7+ is because 3.6 and below are EOL: https://endoflife.date/python

I assume Ubuntu must still be doing security patches for 3.6, but typically packages in the Python ecosystem drop support as soon as a version enters official EOL, so it becomes really hard to work with those versions since all of the core packages no longer support them and you end up having to introduce hacks to force older versions, etc.

For example, just randomly picking some popular packages:

* [pytest requires 3.7+](https://pypi-browser.org/package/pytest/pytest-7.2.1-py3-none-any.whl#:~:text=xmlschema%20%3B%20extra%20%3D%3D%20%27testing%27-,Requires%2DPython,-%3E%3D3.7)
* [setuptools requires 3.7+](https://pypi-browser.org/package/setuptools/setuptools-67.3.2-py3-none-any.whl#:~:text=and%20extra%20%3D%3D%20%27testing%27-,Requires%2DPython,-%3E%3D3.7)
* [tox requires 3.7+](https://pypi-browser.org/package/tox/tox-4.4.5-py3-none-any.whl#:~:text=0.38.4%3B%20extra%20%3D%3D%20%27testing%27-,Requires%2DPython,-%3E%3D3.7)
* [requests requires 3.7+](https://pypi-browser.org/package/requests/requests-2.28.2-py3-none-any.whl#:~:text=3.0.2\)%20%3B%20extra%20%3D%3D%20%27use_chardet_on_py3%27-,Requires%2DPython,-%3E%3D3.7%2C%20%3C4)

Reply:
The issue with requiring 3.7 is that itests on bionic fail because bionic comes with 3.6 instead.

Are you fine with dropping support for bionic, and removing xenial and bionic from the CI tests?

Reply:
yeah, xenial is end-of-lifed and bionic is pretty old at this point

Reply:
I've hit a big issue and a small issue while trying to move `make test` off of bionic. The small issue seems easily fixable, but the big one is enough of a rabbit hole that I'm probably not going to get around to figuring it anytime soon though.

The small issue is that something seems to have changed in how the shells spawned as part of the tests resolve the `aactivator` command. Previously, the tests were able to find `aactivator` just fine, but once I ran it with focal + 3.7, it could no longer find the command. I think the cause is https://github.com/Yelp/aactivator/blob/master/tests/integration_test.py#L34, where there's not a `:` between the `dirname` result and `os.defpath`, resulting in a `PATH` that looks like `/my/venv/bin/bin:/usr/bin` instead of `/my/venv/bin:/bin:/usr/bin`. Adding a `:` between those strings makes it work, but I'm not sure how it worked in the first place.

The bigger issue is that something seems to have changed with the file permission of the `.activate.sh` and `.deactivate.sh` files created during `make test`, resulting in `aactivator` refusing to use them. On focal, the `venv` dir created by tests have a mode of `st_mode=16893` (= `0o40775`), and the `.activate.sh` file has a mode of `st_mode=33204`(= `0o100664`). So aactivator does not activate because the file is group writeable.
But why did this previously work fine in itests? The itests run in docker containers, where the `venv` dir has a mode of `st_mode=16877` (= `0o40755`), and `.activate.sh` file has a mode of `st_mode=33188` (=`0o100644`).
Unfortunately I don't have a bionic machine to compare the non-dockerized test files against.

Anyways, that's all to say that I am not opposed to dropping support for python3.3-3.6 and Ubuntus bionic and below, but I probably won't be the one to do it. I think this PR is sufficient to make aactivator stop warning in recent versions of python, and it's worth merging in separely from updating testing infra to py37 + bionic.